ramp-database/rampdb/model/problem.py

```diff
@@ -32,87 +25,178 @@
-        Integer, ForeignKey('workflows.id'), nullable=False)
-    workflow = relationship(
-        'Workflow', backref=backref('problems'))
+    workflow_id = Column(Integer, ForeignKey('workflows.id'), nullable=False)
+    workflow = relationship('Workflow', backref=backref('problems'))
+
+    # XXX: big change in the database
+    path_ramp_kits = Column(String, nullable=False, unique=False)
+    path_ramp_data = Column(String, nullable=False, unique=False)
 
-    def __init__(self, name):
+    def __init__(self, name, path_ramp_kits, path_ramp_data, session=None):
         self.name = name
-        self.reset()
-        # to check if the module and all required fields are there
-        self.module
-        self.Predictions
-        self.workflow_object
+        self.path_ramp_kits = path_ramp_kits
+        self.path_ramp_data = path_ramp_data
+        self.reset(session)
 
     def __repr__(self):
-        return 'Problem({})\n{}'.format(
-            encode_string(self.name), self.workflow)
-
-    def reset(self):
-        self.workflow = Workflow.query.filter_by(
-            name=type(self.module.workflow).__name__).one()
+        return 'Problem({})\n{}'.format(encode_string(self.name),
+                                        self.workflow)
+
+    def reset(self, session):
+        """Reset the workflow."""
+        if session is not None:
+            self.workflow = \
+                (session.query(Workflow)
+                        .filter(Workflow.name ==
+                                type(self.module.workflow).__name__)
+                        .one())
+        else:
+            self.workflow = \
+                (Workflow.query
+                         .filter_by(name=type(self.module.workflow).__name__)
+                         .one())
 
     @property
     def module(self):
+        """module: Get the problem module."""
         return import_module_from_source(
-            os.path.join(RAMP_KITS_PATH, self.name, 'problem.py'),
+            os.path.join(self.path_ramp_kits, self.name, 'problem.py'),
             'problem'
         )
 
     @property
     def title(self):
+        """str: The title of the problem."""
         return self.module.problem_title
 
     @property
     def Predictions(self):
+        """:class:`rampwf.prediction_types`: The predictions used for the
+        problem."""
         return self.module.Predictions
 
     def get_train_data(self):
-        path = os.path.join(RAMP_DATA_PATH, self.name)
+        """ndarray: The training data."""
+        path = os.path.join(self.path_ramp_data, self.name)
         return self.module.get_train_data(path=path)
 
     def get_test_data(self):
-        path = os.path.join(RAMP_DATA_PATH, self.name)
+        """ndarray: The testing data."""
+        path = os.path.join(self.path_ramp_data, self.name)
         return self.module.get_test_data(path=path)
 
     def ground_truths_train(self):
+        """ndarray: The true labels for the training."""
```

Comment:
The return value here is not a ndarray, but a Predictions instance (and the same fot the two below)